### DISCUSSION TITLE
Importing the remote app component in host app throwing Self-reference dependency 'turbo run build'

### DISCUSSION THREAD

**Discussion by dkaus1** [vercel/turborepo#5722] [Help]
Next.js version. - 12.2.2
Turbo gen version - 1.9.7

tsconfig.json // host app

{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve"
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}

Container:build: Failed to compile.
Container:build:
Container:build: ./node_modules/@module-federation/utilities/src/utils/importDelegatedModule.js
Container:build: Self-reference dependency has unused export name: This should not happen
Container:build:
Container:build: Import trace for requested module:
Container:build: ./node_modules/@module-federation/utilities/src/utils/importDelegatedModule.js
Container:build: ./node_modules/@module-federation/nextjs-mf/src/default-delegate.js?remote=remote_app@http://localhost:3001/_next/static/chunks/remoteEntry.js
Container:build: remote remote_app/page
Container:build:
Container:build: ./node_modules/@module-federation/utilities/src/utils/pure.js
Container:build: Self-reference dependency has unused export name: This should not happen
Container:build:
Container:build: Import trace for requested module:
Container:build: ./node_modules/@module-federation/utilities/src/utils/pure.js
Container:build: ./node_modules/@module-federation/utilities/src/utils/importDelegatedModule.js
Container:build: ./node_modules/@module-federation/nextjs-mf/src/default-delegate.js?remote=remote_app@http://localhost:3001/_next/static/chunks/remoteEntry.js
Container:build: remote remote_app/page

**Comment by tolalitnegi** [comment]
I am also facing the same issue, Vercel team any clue please

**Comment by anthonyshew** [comment]
Could you provide a full reproduction? 🙏
